mlflow/store/rest_store.py

```diff
@@ -107,6 +107,12 @@ def get_experiment(self, experiment_id):
         response_proto = self._call_endpoint(GetExperiment, req_body)
         return Experiment.from_proto(response_proto.experiment)
 
+    def get_experiment_by_name(self, name):
+        for experiment in self.list_experiments():
```

Comment:
Thanks for the PR! Could we pass `ViewType.ALL` to `self.list_experiments` here so that the RestStore matches the existing behavior in the FileStore? I'm specifically comparing this implementation with https://github.com/mlflow/mlflow/blob/master/mlflow/store/file_store.py#L186